nixos/modules/services/hardware/upower.nix

```diff
@@ -179,24 +179,59 @@ in
         '';
       };
 
+      allowRiskyCriticalPowerAction = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        description = ''
+          Enable the risky critical power actions "Suspend" and "Ignore".
+        '';
+      };
+
       criticalPowerAction = lib.mkOption {
-        type = lib.types.enum [ "PowerOff" "Hibernate" "HybridSleep" ];
+        type = lib.types.enum [
+          "PowerOff"
+          "Hibernate"
+          "HybridSleep"
+          "Suspend"
+          "Ignore"
+        ];
         default = "HybridSleep";
         description = ''
           The action to take when `timeAction` or
           `percentageAction` has been reached for the batteries
-          (UPS or laptop batteries) supplying the computer
+          (UPS or laptop batteries) supplying the computer.
+
+          When set to `Suspend` or `Ignore`, `allowRiskyCriticalPowerAction`
+          must be set to `true`.
         '';
       };
 
     };
 
   };
 
-
   ###### implementation
 
   config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion =
+          with builtins;
```

Comment:
nit, feel free to ignore: what is `builtins` used for here? Is it just `elem`? I think this would be nicer as just a `inherit (builtins) elem;` in the let block instead.

Reply:
Done.